### ISSUE TITLE
Varying results for out-of-range casts

### ISSUE THREAD

**Issue by Fleischmaki** [esbmc/esbmc#1392]
Hi, consider the following program (test.c)
```
extern void __VERIFIER_error(void);
extern unsigned int __VERIFIER_nondet_uint(void);

int main(){
        unsigned int n = __VERIFIER_nondet_uint();
        if(!(0 <= (int) (n))){
                if(846 <= n){
                        __VERIFIER_error();
                }
        }
        return 0;
}
```
Running `esbmc bug.c` gives `VERIFICATION FAILED`,
but running `esbmc --interval-analysis bug.c` gives `VERIFICATION_SUCCESFUL`
Not sure if this is intented as the counterxample given is `n = 4294967295` so casting it to signed int gives undefined behaviour iirc. (as the value exceeds the value range for signed integers)
Commit version: e5e00743dceed71bf7a48cc3f34887bf0a0955b8
OS: Ubuntu 20.04

**Comment by rafaelsamenezes** [comment]
Just confirmed it on Integer domain (wrapped domain is fine).

Update: the issue is that we get rid of the typecasts when canonizing the path condition into CNF.

**Comment by lucasccordeiro** [comment]
Closed via #1393.
